src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -4104,10 +4104,11 @@ internal PSModuleInfo LoadRequiredModule(PSModuleInfo currentModule, ModuleSpeci
                     if (currentModule != null)
                     {
                         requiredModules.Add(new ModuleSpecification(currentModule), new List<ModuleSpecification> { requiredModuleSpecification });
+                        requiredModules.Add(requiredModuleSpecification, new List<ModuleSpecification> ( requiredModuleInfo.RequiredModulesSpecification ));
```

Comment:
does requiredModuleSpecification only needs to be added when currentModule is not null?
this is confusing here. if so, please add comment explains why